steel/tests/view_call.rs

```diff
@@ -17,19 +17,34 @@
 use alloy_primitives::{address, b256, uint, Address, U256};
 use alloy_sol_types::{sol, SolCall};
 use risc0_steel::{
-    config::ETH_SEPOLIA_CHAIN_SPEC,
+    config::{ChainSpec, ETH_MAINNET_CHAIN_SPEC, ETH_SEPOLIA_CHAIN_SPEC},
     ethereum::EthViewCallEnv,
-    host::{
-        provider::{CachedProvider, EthFileProvider, EthersProvider},
-        EthersClient,
-    },
-    ViewCall,
+    host, CallBuilder, Contract,
 };
 use std::fmt::Debug;
 use test_log::test;
 
+macro_rules! provider {
```

Comment:
This was the only way I could think of to make the golden cache file generation a bit more streamlined. It also fixes the issue where the spec was not always provided (which was already an issue before this PR).
Happy to change it if there are any better suggestions....